CODING_GUIDELINES.md

```diff
@@ -3,41 +3,41 @@
 
 ## Software design
 
-Designing a clean architecture for a software product is a difficult task, and developers can easily overlook it under the pressure of deadlines or when they are eager to deliver exciting new features. However, choosing the quick solution instead of the clean one will likely lead to a large amount of additional work in the future (_technical debt_).
+Designing a clean architecture for a software product is a difficult task and developers can easily overlook it under pressure or when they are eager to deliver exciting new features. However, choosing the quick solution instead of a clean one will likely lead to a large amount of additional work in the future (_technical debt_).
 
-To keep technical debt at acceptable levels, follow best practices when designing and implementing new features:
+To keep technical debt at acceptable levels, design and implementation of new features should follow best practices:
 
-1. Make sure all your code is covered by automatic testing to ensure its correctness. Where unit tests are impractical, use integration tests.
+1. Make sure your code is covered by automatic testing to ensure its correctness. Where unit tests are impractical, use integration tests.
 2. Adhere to the [SOLID](https://en.wikipedia.org/wiki/SOLID) principles of software design.
-3. Do not reinvent the wheel: if someone else solved your problem within the project or in a third party library, consider using it or extending it before writing a new component for basically the same purpose.
-4. _You ain't gonna need it_ ([YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)): don't design solutions for problems that might come up in the future, chances are you will never need that code. Focus on the current problems and prepare for future requirements by writing clean code.
-5. _Do not repeat yourself_ ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)): if you are writing down the same code pattern in several places you should extract it into a function.
-6. Use meaningful names: the purpose of an object should be clear from its name. Class names are usually nouns and function names are verbs.
+3. Do not reinvent the wheel: if someone solved your problem within the project or in a third party library, consider using their solution or extending it before writing a new component for the same purpose.
+4. _You aren't gonna need it_ ([YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)): do not design solutions for problems that might come up in the future, as chances are that you will never need that code. Focus on current problems and prepare for future requirements by writing clean code.
```

Comment:
I think the original phrasing is still the most common way to refer to this idiom:
```suggestion
4. _You ain't gonna need it_ ([YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)): do not design solutions for problems that might come up in the future, as chances are that you will never need that code. Focus on current problems and prepare for future requirements by writing clean code.
```